From 8163bd131716118b73e011ce46c6234787058b24 Mon Sep 17 00:00:00 2001
From: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Date: Wed, 13 May 2020 20:47:01 +0530
Subject: [PATCH 1359/1851] dmaengine: xilinx: axidmatest: Fix variable length
 array usage
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use constant length array. Fixes below gcc warning-

drivers/dma/xilinx/axidmatest.c: In function ‘dmatest_slave_func’:
drivers/dma/xilinx/axidmatest.c:294:3: warning: ISO C90 forbids variable
length array 'dma_srcs' [-Wvla]
dma_addr_t dma_srcs[src_cnt];
   ^~~~~~~~~~
drivers/dma/xilinx/axidmatest.c:295:3: warning: ISO C90 forbids variable
length array 'dma_dsts' [-Wvla]
   dma_addr_t dma_dsts[dst_cnt];
   ^~~~~~~~~~
drivers/dma/xilinx/axidmatest.c:302:10: warning: ISO C90 forbids variable
length array 'tx_sg' [-Wvla]
   struct scatterlist tx_sg[bd_cnt];
          ^~~~~~~~~~~
drivers/dma/xilinx/axidmatest.c:303:10: warning: ISO C90 forbids variable
length array 'rx_sg' [-Wvla]
   struct scatterlist rx_sg[bd_cnt];

Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/axidmatest.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/dma/xilinx/axidmatest.c b/drivers/dma/xilinx/axidmatest.c
index 63ef98510712..c60c214bf4ae 100644
--- a/drivers/dma/xilinx/axidmatest.c
+++ b/drivers/dma/xilinx/axidmatest.c
@@ -49,6 +49,8 @@ MODULE_PARM_DESC(iterations,
 #define PATTERN_OVERWRITE	0x20
 #define PATTERN_COUNT_MASK	0x1f
 
+#define XILINX_DMATEST_BD_CNT	11
+
 struct dmatest_slave_thread {
 	struct list_head node;
 	struct task_struct *task;
@@ -241,7 +243,7 @@ static int dmatest_slave_func(void *data)
 	int ret;
 	int src_cnt;
 	int dst_cnt;
-	int bd_cnt = 11;
+	int bd_cnt = XILINX_DMATEST_BD_CNT;
 	int i;
 
 	ktime_t	ktime, start, diff;
@@ -291,16 +293,16 @@ static int dmatest_slave_func(void *data)
 		struct dma_device *rx_dev = rx_chan->device;
 		struct dma_async_tx_descriptor *txd = NULL;
 		struct dma_async_tx_descriptor *rxd = NULL;
-		dma_addr_t dma_srcs[src_cnt];
-		dma_addr_t dma_dsts[dst_cnt];
+		dma_addr_t dma_srcs[XILINX_DMATEST_BD_CNT];
+		dma_addr_t dma_dsts[XILINX_DMATEST_BD_CNT];
 		struct completion rx_cmp;
 		struct completion tx_cmp;
 		unsigned long rx_tmo =
 				msecs_to_jiffies(300000); /* RX takes longer */
 		unsigned long tx_tmo = msecs_to_jiffies(30000);
 		u8 align = 0;
-		struct scatterlist tx_sg[bd_cnt];
-		struct scatterlist rx_sg[bd_cnt];
+		struct scatterlist tx_sg[XILINX_DMATEST_BD_CNT];
+		struct scatterlist rx_sg[XILINX_DMATEST_BD_CNT];
 
 		total_tests++;
 
-- 
2.31.1

